### PR TITLE
Adding take_last.

### PR DESCRIPTION
**Pull request by BRevzin**
Implements #1162, `take_last` simply calls `drop_exactly` and is only supported for `SizedRange`s. 

If you pass in a size larger than the input size, I just round down.

### CODE REVIEW

File: include/range/v3/view/take_last.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    namespace view
+    {
+        struct take_last_fn
+        {
+        private:
+            friend view_access;
+            template<typename Int>
+            static auto CPP_fun(bind)(take_last_fn take_last, Int n)( //
+                requires Integral<int>)
```
Comment (timsong-cpp) [inline include/range/v3/view/take_last.hpp]: Int, not int.

File: doc/index.md [hunk 2]
```diff
@@ -763,6 +763,8 @@ provides, and a blurb about how each is intended to be used.
   <DD>Given a source range and an integral count, return a range consisting of the first *count* elements from the source range, or the complete range if it has fewer elements. (The result of `view::take` is not a `SizedRange`.)</DD>
 <DT>\link ranges::view::take_exactly_fn `view::take_exactly`\endlink</DT>
   <DD>Given a source range and an integral count, return a range consisting of the first *count* elements from the source range. The source range must have at least that many elements. (The result of `view::take_exactly` is a `SizedRange`.)</DD>
+<DT>\link ranges::view::drop_exactly_fn `view::take_last`\endlink</DT>
```
Comment (johelegp) [inline doc/index.md]: There's a `drop` here that should be changed to `take`.

Comment (BRevzin) [reply doc/index.md]: This one's right - there's not actually a `take_last_fn` in this implementation, `take_last` just calls `drop_exactly`. Unless it should like somewhere else?

Comment (johelegp) [reply doc/index.md]: The `_fn` are the range adaptors, which must exist, unlike the `_view`.

Comment (BRevzin) [reply doc/index.md]: Oh, I misunderstood everything.

Comment (johelegp) [reply doc/index.md]: I must correct myself. The range adaptors are the `_fn` wrapped in `view::view<>`.

**Comment by BRevzin** [comment]
retest this please

**Comment by CaseyCarter** [comment]
LGTM. @ericniebler, do you want to perform design sign-off on this feature addition? Most notably, do we want to take this as-is or require support for forward non-sized ranges?

File: include/range/v3/view/take_last.hpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            {
+                return make_pipeable(std::bind(take_last, std::placeholders::_1, n));
+            }
+
+        public:
+            template<typename Rng>
+            auto CPP_fun(operator())(Rng && rng, range_difference_t<Rng> n)(
+                const requires ViewableRange<Rng> && SizedRange<Rng>)
+            {
+                auto sz = ranges::distance(rng);
```
Comment (ericniebler) [inline include/range/v3/view/take_last.hpp]: Why is this `distance` and not `size`?

Comment (BRevzin) [reply include/range/v3/view/take_last.hpp]: signed vs unsigned.

**Pull request merged**